components/src/histogram.tsx

```diff
@@ -19,18 +19,21 @@
  *
  */
 
-import iobioCharts from 'iobio-charts';
-import { useEffect } from 'react';
-
-const { createHistogram } = iobioCharts;
-
-function Histogram(props: { 'broker-key': string }) {
-	useEffect(() => {
-		createHistogram();
-	}, []);
-
-	// return <iobio-histogram />;
-	return <div broker-key={props['broker-key']}></div>;
+function IobioHistogram({
+	brokerKey,
+	color,
+	ignoreOutliers,
+	title,
+}: {
+	brokerKey: string;
+	color?: string;
+	ignoreOutliers?: boolean;
+	title?: string;
+}) {
+	const outlierProp = ignoreOutliers ? { 'ignore-outliers': true } : {};
```

Comment:
I think this is ok for now. I don't see the need for `{}` object.
Good opportunity to write something for boolean html attributes if you think we'd handle more of them.
a function that returns a list of strings, then spread the string list on the web component

Reply:
I see what you're saying but in practice this works: 
`const outlierProp = ignoreOutliers ? { 'ignore-outliers': true } : {};`

( {} is so we can do `{...outlierProp}` )

this does not: 
`const outlierProp = ignoreOutliers ? ['ignore-outliers'] : [];`
with error `Invalid attribute name: '0'`, so it is treating the array as key/value rather than just dropping the value in as a property

Reply:
It could be a util to create `{ [string]: true }` for all similar props but currently this is the only boolean attribute we're using

Reply:
You're right, array won't spread as strings but this is along the lines of what I was thinking.

```js 
// func
const getBooleanAttributes = (booleanAttributesForThisComponent) => {
	return Object.keys(booleanAttributesForThisComponent)
		.filter((k) => booleanAttributesForThisComponent[k])
		.reduce((acc, key) => {
			return { ...acc, [key]: booleanAttributesForThisComponent[key] };
		}, {});
};
```

```js
// in component
	const booleanAttributesForThisComponent = {
		'ignore-outliers': ignoreOutliers,
		'show-bam': true,
	};

	const booleanAttributes = getBooleanAttributes(booleanAttributesForThisComponent);

	return (
		<iobio-histogram
			broker-key={brokerKey}
			title={title}
			style={{ '--iobio-data-color': color }}
			{...booleanAttributes}
		/>
	);
	```

Reply:
OK I've added this. Limited use at the moment but it's there if we need it.

Reply:
Thank you for adding this.

I'm sorry if I wasn't clear that my code snippet was a starting point - my code would not pass PR review.

If it's worth adding, it's worth adding right. Or wrap up this PR and have it as an improvement later on.

https://github.com/overture-stack/iobio-components/pull/24/commits/c8075142b994d6f99f5c86b317205f28f9582151#r1666912884
https://github.com/overture-stack/iobio-components/pull/24#discussion_r1666915603